src/bun.js/bindings/ZigGlobalObject.cpp

```diff
@@ -479,6 +480,15 @@ extern "C" JSC__JSGlobalObject* Zig__GlobalObject__create(void* console_client,
 
     JSC::gcProtect(globalObject);
 
+    vm.setOnEachMicrotaskTick([globalObject](JSC::VM& vm) -> void {
+        if (auto nextTickQueue = globalObject->m_nextTickQueue.get()) {
+            vm.setOnEachMicrotaskTick(nullptr);
```

Comment:
why is the handler cleared here? it doesn't seem to ever be restored.

also we now need to coordinate this with the async_hooks `setOnEachMicrotaskTick` overwriting this handler temporarily.

I will push some tests that hit these edge cases